Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/ServicePrincipalAzureManager.java

```diff
@@ -198,55 +78,65 @@ public KeyVaultClient getKeyVaultClient(String tid) {
             @Override
             public String doAuthenticate(String authorization, String resource, String scope) {
                 try {
-                    initATCIfNeeded();
-                    return atc.getToken(resource);
+                    return ServicePrincipalAzureManager.this.credentials.getToken(resource);
                 } catch (Exception ex) {
                     throw new RuntimeException(ex);
                 }
             }
         };
-
         return new KeyVaultClient(creds);
     }
 
     @Override
-    public String getCurrentUserId() throws IOException {
-        initATCIfNeeded();
-        return atc.clientId();
+    public String getAccessToken(String tid, String resource, PromptBehavior promptBehavior) throws IOException {
+        String uri = getManagementURI();
+        return this.credentials.getToken(uri);
     }
 
     @Override
-    public String getAccessToken(String tid, String ignored1, PromptBehavior ignored2) throws IOException {
-        String uri = getManagementURI();
-        return atc.getToken(uri);
+    public String getCurrentUserId() throws IOException {
+        return this.credentials.clientId();
     }
 
     @Override
-    public String getManagementURI() throws IOException {
-        initATCIfNeeded();
-        // default to global cloud
-        return atc.environment() == null ? AzureEnvironment.AZURE.resourceManagerEndpoint() : atc.environment().resourceManagerEndpoint();
+    protected String getDefaultTenantId() throws IOException {
+        return this.defaultTenantId;
     }
 
     @Override
-    public Environment getEnvironment() {
-        initEnv();
-        return env;
+    protected AzureTokenCredentials getCredentials(String tenantId) {
+        return this.credentials;
+    }
+
+    protected boolean isSignedIn() {
+        return Objects.nonNull(this.credentials) && Objects.nonNull(this.defaultTenantId);
     }
 
     @Override
-    public String getStorageEndpointSuffix() {
-        return getEnvironment().getAzureEnvironment().storageEndpointSuffix();
+    public String getManagementURI() throws IOException {
+        // default to global cloud
+        return this.credentials.environment() == null ?
+                AzureEnvironment.AZURE.resourceManagerEndpoint() : this.credentials.environment().resourceManagerEndpoint();
     }
 
-    private void initATCIfNeeded() throws IOException {
-        if (atc == null) {
-            atc = ApplicationTokenCredentials.fromFile(credFile);
+    private void init() {
+        if (this.credentials == null && credFile != null) {
+            try {
+                this.credentials = ApplicationTokenCredentials.fromFile(credFile);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
avoid to convert checked exception to unchecked ones, use AzureExecutionException